terraform/iam.tf

```diff
@@ -61,13 +61,15 @@ resource "google_project_iam_custom_role" "terraform_service_account_role" {
 }
 
 resource "google_project_iam_member" "terraform_service_account_permissions" {
-  project = var.project
-  role    = google_project_iam_custom_role.terraform_service_account_role.id
-  member  = "serviceAccount:${var.terraform_svc_account_email}"
+  depends_on = [google_project_iam_custom_role.terraform_service_account_role]
```

Comment:
this should not be necessary. The `role = ` assignment already refers to this resource so the hint already exists.